go/pkg/sysdb/metastore/db/dao/collection.go

```diff
@@ -209,6 +209,31 @@ func (s *collectionDb) Update(in *dbmodel.Collection) error {
 	return nil
 }
 
+func (s *collectionDb) UpdateLogPositionAndVersionInfo(
+	collectionID string,
+	logPosition int64,
+	currentCollectionVersion int32,
+	currentVersionFileName string,
+	newCollectionVersion int32,
+	newVersionFileName string,
+) (int64, error) {
+	result := s.db.Model(&dbmodel.Collection{}).
+		Clauses(clause.Locking{Strength: "UPDATE"}).
+		Where("id = ? AND version = ? AND version_file_name = ?",
```

Comment:
out of curiosity, do you need the Clauses(clause.Locking{Strength: "UPDATE"}) here? The where id = '?' clause restricts the update set to only one row hence there is no possibility of concurrent updates racing with this update right for the duration of this update?